### ISSUE TITLE
[css-text] Clarify importance of writing system as well as language for lang-specific effects

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#2015]
https://www.w3.org/TR/css-text-3/#languages notes that typographic effects vary by language but should also mention writing system, and should give examples like ja-latn to emphasize that UAs need to detect script tags. See https://twitter.com/little_potato/status/935608512297820160

**Comment by frivoal** [comment]
@fantasai I wonder if this edit is actually sufficient. Shouldn't we also change bits of text like that in the definition of the line break property which says things like "do FOOBAR if the content language is Chinese or Japanese", and instead say "do FOOBAR if the **writing system** is Chinese or Japanese", liking to the text that https://github.com/w3c/csswg-drafts/commit/c983ca60b3a268fefd93f81d81495974c8eb2484 added to point out that the writing system should be assumed to be that of the content language, **unless** the author has said otherwise?

If we don't do that, it seems to be that the text you've added, while a great explanation of the situation, isn't actually invoked properly from the rest of the spec.

**Comment by frivoal** [comment]
Reopening, so that the above comment doesn't get lost. We can close again if there's nothing to be done, but I'd like to make sure.

**Comment by frivoal** [comment]
But can we actually say "if the writing system is Chinese"? Chinese isn't a writing system, so it seems we should we instead say something like "if the writing system is one of Hans / Hant / Hani / Hanb, or the language is zh and the writing system is unspecified". It would be nice if we could drop the second part of the sentence, but it's not clear to me that the mapping from languages to default writing-system for that language is fully defined, and if it isn't, then we need that second clause.

**Comment by frivoal** [comment]
Then again, the above introduces a direct dependency from css to [iso15924](http://unicode.org/iso15924/iso15924-codes.html) writing system tags, instead of leaving the specific way to encode these to the document language. So maybe we should speak of the "Chinese writing system", rather than "one of Hans / Hant / Hani / Hanb"…

**Comment by acli** [comment]
Since I was the one making the original complaint, I should first apologize for not joining in the discussion, but I think I need to make it clear that my issue was not specific to the Chinese writing system. I did observe the problem while marking up Chinese, but I can’t see how this isn’t going to affect other languages that can be written in multiple scripts (basically anything that’s not written in the Latin alphabet).

I just cobbled up a minimal test:

    <!doctype html>
    <html lang=en>
    <head>
    <meta charset=utf-8>
    <title>Test</title>
    </head>
    <body>
    <p>
    This is a sentence with a Chinese word, <span lang=zh-yue-latn>ping4 on1</span>;
    an Arabic word, <span lang=ar-latn>salaam</span>;
    and a Hebrew word, <span lang=he-latn>shalom</span>.
    </body>
    </html>

I have not tried overriding any styling yet, but with default styling all the three non-English words are being displayed using a different font (on Firefox Nightly 64.0a1) even though they are all “latn”. This doesn’t look like something specific to the Chinese writing system to me.
